### DISCUSSION TITLE
Adding a new infix to tokenizer distorts tokenizer's previous behavior with scispaCy

### DISCUSSION THREAD

**Discussion by mertugrul** [explosion/spaCy#8362] [Help: Coding & Implementations] [answered]
I am using 'en_core_sci_md' for processing some clinical reports. Its tokenizer currently does not consider "/" and "-" as infix. I'd like to keep "-" as non-infix, but add "/" to the infixes while tokenizing the reports.  When I added "/" as infix, the behavior of the tokenizer is distorted. '-' is also added as an infix (I think it is because of nlp.Defaults.infixes). But, I need to keep it as non-infix. Is there any way to add a new infix while keeping the existing tokenizer's behavior same (e.g. other infixes, prefixes, suffixes)? 

`import spacy`
`import scispacy`

`nlp = spacy.load("en_core_sci_md")`

`text = "New report related to recursive/residual disease of para-thyroid."`

`[token.text for token in nlp(text)]`
['New', 'report', 'related', 'to', 'recursive/residual', 'disease', 'of', 'para-thyroid']

`infixes = nlp.Defaults.infixes + ["/"]`
`infix_regex = spacy.util.compile_infix_regex(infixes)`
`nlp.tokenizer.infix_finditer = infix_regex.finditer`

`[token.text for token in nlp(text)]`
['New', 'report', 'related', 'to', recursive', '/', 'residual', 'disease', 'of', 'para', '-', 'thyroid']

## Your Environment

* Operating System: Ubuntu 18.04
* Python Version Used: 3.8.8
* spaCy Version Used: 3.0.6
* Environment Information:

**Comment by svlandeg** [comment, marked as answer]
You can find the original tokenization rules for sciscpacy here: https://github.com/allenai/scispacy/blob/main/scispacy/custom_tokenizer.py

As you've found, just overwriting `infixes` will remove all the custom infix rules that are defined there. So it would make sense to copy the infix parts from that code, add what you want, and then use that to customize your tokenization.

**Reply by mertugrul** [reply]
Thank you, that resolved my problem.
